src/app/features/creature/npc-vendor/npc-vendor.service.ts

```diff
@@ -1,19 +1,72 @@
 import { Injectable } from '@angular/core';
-import { ToastrService } from 'ngx-toastr';
-
 import { MultiRowEditorService } from '@keira-abstract/service/editors/multi-row-editor.service';
-import { CreatureHandlerService } from '../creature-handler.service';
+import { IconService } from '@keira-shared/modules/icon/icon.service';
 import { MysqlQueryService } from '@keira-shared/services/mysql-query.service';
-import { NPC_VENDOR_ID, NPC_VENDOR_ID_2, NPC_VENDOR_TABLE, NpcVendor } from '@keira-types/npc-vendor.type';
+import { SqliteQueryService } from '@keira-shared/services/sqlite-query.service';
+import { ItemExtendedCost } from '@keira-shared/types/item-extended-cost.type';
+import { NpcVendor, NPC_VENDOR_ID, NPC_VENDOR_ID_2, NPC_VENDOR_TABLE } from '@keira-types/npc-vendor.type';
+import { ToastrService } from 'ngx-toastr';
+import { CreatureHandlerService } from '../creature-handler.service';
 
 @Injectable()
 export class NpcVendorService extends MultiRowEditorService<NpcVendor> {
   /* istanbul ignore next */ // because of: https://github.com/gotwarlost/istanbul/issues/690
   constructor(
     protected handlerService: CreatureHandlerService,
     public readonly queryService: MysqlQueryService,
+    public readonly sqliteQueryService: SqliteQueryService,
     protected toastrService: ToastrService,
+    private iconService: IconService,
   ) {
     super(NpcVendor, NPC_VENDOR_TABLE, NPC_VENDOR_ID, NPC_VENDOR_ID_2, handlerService, queryService, toastrService);
   }
+
+  private cache: { [key: string]: Promise<string>[] } = {};
```

Comment:
I think one array of `Promise<string>` is enough, we don't need the object with key